pyfr/integrators/dual/pseudo/multip.py

```diff
@@ -23,6 +24,8 @@ def __init__(self, backend, systemcls, rallocs, mesh, initsoln, cfg,
 
         # Get the multigrid cycle
         self.cycle, self.csteps = zip(*cfg.getliteral(mgsect, 'cycle'))
+        random.seed(cfg.getint(mgsect, 'seed', 0))
```

Comment:
Changing the global seed is not appropriate.  Instead create a NumPy generator object.